StitchCore/StitchCore/Source/Network/StitchTask.swift

```diff
@@ -1,129 +1,337 @@
 import Foundation
 
-/// Promise system to handle requests to Stitch
-open class StitchTask<Result> {
+/// Generic results structure that contains success and failure values.
+public enum StitchResult<Value> {
+    /// If the Task was successful with a value
+    case success(Value)
+    /// If the Task failed with a value
+    case failure(Error)
     
-    private let queue: OperationQueue
+    /// Value from the `StitchTask` in the event of success
+    public var value: Value? {
+        switch self {
+        case .success(let value):
+            return value
+        case .failure:
+            return nil
+        }
+    }
     
-    /// The final result of this task
-    public var result: StitchResult<Result>? {
-        didSet{
-            queue.isSuspended = false
+    /// Error from the `StitchTask` in the event of failure
+    public var error: Error? {
+        switch self {
+        case .success:
+            return nil
+        case .failure(let error):
+            return error
         }
     }
+}
+
+public protocol ExecutionContext {
+    func execute(_ work: @escaping () -> Void)
+}
+
+extension DispatchQueue: ExecutionContext {
+    public func execute(_ work: @escaping () -> Void) {
+        self.async(execute: work)
+    }
+}
+
+public final class InvalidatableQueue: ExecutionContext {
```

Comment:
It can be used as an `ExecutionContext` by the consumer of the SDK. There are unused methods in both mobile SDKs with the intent that a consumer may use them. 

As long as we are doing our own promise impl, it might be a good idea to keep around.